src/runtime_src/core/tools/xbmgmt2/flash/flasher.cpp

```diff
@@ -36,43 +36,54 @@
 #define FLASH_BASE_ADDRESS BPI_FLASH_OFFSET
 #define MAGIC_XLNX_STRING "xlnx" // from xclfeatures.h FeatureRomHeader
 
-Flasher::E_FlasherType Flasher::getFlashType(std::string typeStr)
-{
-    std::string err;
-    E_FlasherType type = E_FlasherType::UNKNOWN;
-    if (typeStr.empty())
-      typeStr = xrt_core::device_query<xrt_core::query::f_flash_type>(m_device);
-    if (typeStr.empty())
-      typeStr = xrt_core::device_query<xrt_core::query::flash_type>(m_device);
-
-    if (typeStr.empty())
-    {
-        getProgrammingTypeFromDeviceName(mFRHeader.VBNVName, type);
-    }
-    else if (typeStr.compare("spi") == 0)
-    {
+/* 
+ * map flash_tyoe str to E_FlasherType
+ */
+void Flasher::typeStr_to_E_FlasherType(const std::string& typeStr, E_FlasherType &type)
+{    
+    if (typeStr.compare("spi") == 0) {
         type = E_FlasherType::SPI;
     }
-    else if (typeStr.compare("bpi") == 0)
-    {
+    else if (typeStr.compare("bpi") == 0) {
         type = E_FlasherType::BPI;
     }
-    else if (typeStr.find("qspi_ps") == 0)
-    {
+    else if (typeStr.find("qspi_ps") == 0) {
         // Use find() for this type of flash.
         // Since it have variations
         type = E_FlasherType::QSPIPS;
     }
-    else if (typeStr.compare("ospi_versal") == 0)
-    {
+    else if (typeStr.compare("ospi_versal") == 0) {
         type = E_FlasherType::OSPIVERSAL;
     }
-    else
-    {
-        std::cout << "Unknown flash type: " << typeStr << std::endl;
+    else {
```

Comment:
Being that the enum E_FlasherType has an UNKNOWN value, please return this value instead of throwing an exception.  

It will be the responsibility of the calling code to determine if an exception needs to be thrown.